src/common/src/consistent_hash/vnode.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// `VirtualNode` is the logical key for consistent hash. Virtual nodes stand for the intermediate
+/// layer between data and physical nodes.
+pub type VirtualNode = u16;
+pub const VNODE_BITS: usize = 11;
+pub const VIRTUAL_NODE_COUNT: usize = 1 << VNODE_BITS;
+pub const VNODE_BITMAP_LEN: usize = 1 << (VNODE_BITS - 3);
+
+// `VNodeBitmap` is a bitmap of vnodes for a state table, which indicates the vnodes that the state
+// table owns.
+#[derive(Clone, Default)]
+pub struct VNodeBitmap {
+    table_id: u32,
+    bitmap: Vec<u8>,
+}
+
+impl From<risingwave_pb::common::VNodeBitmap> for VNodeBitmap {
+    fn from(proto: risingwave_pb::common::VNodeBitmap) -> Self {
+        Self {
+            table_id: proto.table_id,
+            bitmap: proto.bitmap,
+        }
+    }
+}
+
+impl VNodeBitmap {
+    pub fn new(table_id: u32, bitmap: Vec<u8>) -> Self {
+        Self { table_id, bitmap }
+    }
+
+    pub fn check_overlap(&self, bitmaps: &[risingwave_pb::common::VNodeBitmap]) -> bool {
+        if bitmaps.is_empty() {
+            return true;
+        }
+        if let Ok(pos) =
+            bitmaps.binary_search_by_key(&self.table_id, |bitmap| bitmap.get_table_id())
+        {
+            let text = &bitmaps[pos];
```

Comment:
a small question: why it's called "text"? 👀

Reply:
Just copied the function from `store.rs`. 🤣 Weird indeed. Lemme change it.